### PR TITLE
Secbots are louder and have a delay before they attack

### PR DESCRIPTION
**Pull request by Superlagg**
## About the PR 

Secbots no longer instantly attack the moment they get in range of whatever they want to kill. Instead, they stop, spend a couple seconds to charge their baton, then try to stun their target. If that person is still right next to them when their thing is charged, they'll stun them, but if not, they'll chase after their target, and if they reach them within a second, they instantly zap them.

The secbot siren thing has been moved to its own proc, and is called in several places when in pursuit. Now there's no doubt about whether Beepsky's running someone down!

Also, non-beepsky secbots take 8 seconds to cuff someone, up from 6.

## Why's this needed? 

Secbots were awful to fight, and I think that their role should be to force badguys to keep moving, as opposed to outright catching them. Also, I like obnoxious noises.

## Changelog

```
(u)Superlagg:
(*)Secbots no longer instantly zap you the moment they get in range. Instead, they take a moment to charge their baton, then zap you!
(+)Every secbot, except for Officer Beepsky, forgot to attend mandatory handcuff training, and thus now take 8 seconds to cuff someone.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
Wanna also make secbots absorb weapons fire? Or at least when they get hit by the projectile? Not that it's particularly a nerf like is, but I've always felt like they should because shooting at them feels unresponsive.

File: code/modules/robotics/bot/secbot.dm [hunk 1]
```diff
@@ -93,6 +96,10 @@
 
 	var/last_attack = 0
 	var/attack_per_step = 0 // Tries to attack every step. 1 = 75% chance to attack, 2 = 25% chance to attack
+	var/weeooing /// One WEEOOWEEOO at a time, please
+	var/baton_charged /// Set by the stun action bar if the target isnt in range, grants a brief window for a free zap next time they try to attack
+	var/baton_charge_duration = BATON_CHARGE_DURATION /// How long these batons hold a charge
+	var/cuffing /// So we dont try to cuff someone while we're cuffing someone
```
[the remark this answers is not in the document]
Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm]: Done

**Comment by Superlagg** [comment]
> Wanna also make secbots absorb weapons fire? Or at least when they get hit by the projectile? Not that it's particularly a nerf like is, but I've always felt like they should because shooting at them feels unresponsive.

Tried doing something like that, then I got distracted and made a bunch of projectile mob piercing code that accomplished this in a roundabout way. I'll probably implement that in a later PR, since it'll have to end up messing with projectile code if I don't want to make secbots dense.

File: code/modules/robotics/bot/secbot.dm [hunk 2]
```diff
@@ -419,53 +431,63 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
...
+			SPAWN_DBG(0.2 SECONDS)
 				src.icon_state = "secbot[src.on][(src.on && src.emagged >= 2) ? "-wild" : null]"
-				return
+			if (src.target.getStatusDuration("weakened"))
+				src.mode = SECBOT_PREP_ARREST
+				src.anchored = 1
+				src.target_lastloc = M.loc
+				src.moving = 0
+
+				//qdel(src.mover)
```
Comment (ZeWaka) [inline code/modules/robotics/bot/secbot.dm:481]:
```suggestion
```

**Review by ZeWaka — APPROVED** [review]
looks pretty good

File: code/modules/robotics/bot/secbot.dm [hunk 3]
```diff
@@ -763,37 +786,29 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
...
-					light.set_brightness(0.4)
-					light.set_color(1, 1, 1)
-					*/
-
-					add_simple_light("secbot", list(255, 255, 255, 0.4 * 255))
 				break
 			else
 				continue
 
+	proc/weeoo()
```
Comment (Tarmunora) [inline code/modules/robotics/bot/secbot.dm:795]: 😕

**Review by Tarmunora — APPROVED** [review]

File: code/modules/robotics/bot/secbot.dm (hunk 3 above)
Comment (ZeWaka) [reply code/modules/robotics/bot/secbot.dm:794]: should be named weewoo

**Pull request merged**